_source/_posts/2022-03-29-github-actions-graalvm.adoc

```diff
@@ -0,0 +1,490 @@
+---
+layout: blog_post
+title: "Use GitHub Actions to Build GraalVM Native Images"
+author: matt-raible
+by: advocate
+communities: [java,devops]
+description: "Sometimes it takes a journey of many hours and many builds to produce a released artifact. GraalVM + GitHub Actions = ❤️."
+tags: [graalvm,jhipster,springnative,jhipsternative]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+// keywords: github actions (74K), graalvm spring boot (590), github actions tutorial (2400), github ci cd (5400)
+
+Getting something to work is one of the greatest feelings you can have as a developer. Especially when you've spent hours, days, or months trying to make it happen. The last mile can be one of the most painful and rewarding experiences, all wrapped into the same day or two.
+
+I experienced this recently with link:/blog/2022/03/03/spring-native-jhipster[Spring Native for JHipster]. If I look back, it took a year's worth of desire, research, and perseverance to make it happen. When we finally got it working&mdash;and automated&mdash;you can image my excitement!
+
+After a few days of euphoria, I thought it'd be easy to create native builds for each operating system (Linux, macOS, and Windows) using GitHub Actions. I was wrong.
+
+If you'd like to follow along and learn how to configure GitHub Actions to create native binaries, you'll need a few prerequisites.
+
+*Prerequisites*:
+
+- https://sdkman.io[Java 11]+
+- https://nodejs.org[Node 16]
+- A https://github.com/signup[GitHub Account]
+
+toc::[]
+
+You can read the full conversation (aka a condensed version of this post) via the tweet below, or keep reading to walk through the trials and tribulations I experienced with GitHub Actions and GraalVM. Hopefully my learnings will save you hours of time.
+
+++++
+{% twitter 1496278460293324801 %}
+++++
+
+== Configure a Full-Stack Native Java app to use GitHub Actions
```

Comment:
```suggestion
== Configure a full-stack native Java app to use GitHub Actions
```
quibble, quibble

Reply:
I also had a couple of other ideas in the comments below.